locksmith/src/graphql/datasource/keyholdersByLock.ts

```diff
@@ -32,12 +33,12 @@ export class KeyHoldersByLock extends UnlockGraphQLDataSource {
     const queryResults = await this.getKeyHolders(lockAddress)
 
     try {
-      if (queryResults.data.locks === 0) {
+      if (queryResults.length === 0) {
         return []
       }
-      return queryResults.data.locks
-        .map((lock: any) => lock.keys)[0]
-        .map((key: any) => key.owner.address)
+      return queryResults[0].keys.map((key: any) =>
```

Comment:
Actually, I think we can fix that query in a better way

Reply:
I am confused by the `[0]` here?

Reply:
It retrieves the first and only element of the arrray

Reply:
that I think I understand ;p
What I am confused about is what this element is? is that "per lock"?

Reply:
The graph always returns a collection. In this case, because we are trying to get the keyholders for a particular lock, we can have a collection with either 0 or 1 elements (no lock found with that address/an error occurred and we will treat that as not having found a lock, or exactly one lock found with that address).

So when we want to operate on that lock, we have to pull it out of the array. Otherwise mapping over a collection of one lock so that we can map over its keys would make the return result a nested array, which would complicate usage at call sites.

We can make changes to make this support multiple locks, but for that to be beneficial we would have to make changes elsewhere in the stack as well which I think is not strictly necessary at the moment.

Reply:
ok!